### ISSUE TITLE
pyuvc for Windows

### ISSUE THREAD

**Issue by flecheria** [pupil-labs/pupil#153]
Hi, I'm compiling Pupil in Windows 7 but I get this error:

```
c:\workspace\pupil\pupil_src\capture>python main.py

Traceback (most recent call last):
  File "main.py", line 80, in <module>
    from eye import eye
  File "c:\workspace\pupil\pupil_src\capture\eye.py", line 37, in <module>
    from video_capture import autoCreateCapture, FileCaptureError, EndofVideoFileError, CameraCaptureError
  File "c:\workspace\pupil\pupil_src\shared_modules\video_capture\__init__.py",
line 39, in <module>
    from uvc_capture import Camera_Capture,device_list,CameraCaptureError
  File "c:\workspace\pupil\pupil_src\shared_modules\video_capture\uvc_capture.py", line 11, in <module>
    import uvc
ImportError: No module named uvc
```

Where uvc modules is supposed to be? Do I have to install the module from somewhere?
Thanks
Paolo

**Comment by mkassner** [comment]
@MichaelBarz Whats the status in pyuvc for windows?

**Comment by flecheria** [comment]
this is exactly the problem, I don't have pyuvc.
What do you mean with "status"?

**Comment by flecheria** [comment]
but what is pyuvc?

**Comment by flecheria** [comment]
Ok I find https://github.com/pupil-labs/pyuvc.
I miss completely this library. it's necessary to update the building instractions.
libusb must be 64 bit?

**Comment by MichaelBarz** [comment]
I had a short look at this topic as I concentrated on PyAV for the Player recently.
The build instructions for Windows are not complete, because we never built it for that platform.
But you are right, you have to use x64 versions of all dependencies.

It would be great if you can share you experiences either success or fail.
We'll try to build the lib for Windows in parallel and report our progress either.

Best, Michael

**Comment by flecheria** [comment]
I have to say that Pupil is a very technical project for my skills.
I solved a lot of problems with windows, expecially to build the 64bit of Python.

But now I'm stuck with pyuvc.
I understandand that you have code a unified system to manage the cameras that use libusb, libuvc, jpegturbo and pyuvc above all... a lot of work to do :-)

**Comment by mkassner** [comment]
Hi,

yes, this is still work in progress. You can either:

- help us get it running or 
- wait a bit for Michael to figure it out and update the build instructions or 
- check out a prior version that used our old windows capture backend. Then you dont need pyuvc.

**Comment by rokigeorg** [comment]
hi,

I got the same error message like flecheria. 
But I was concerned about the x64bit.

Is the the pyuvc module only on x64bit systems running?
Is it possible to run the pyuvc module on a Raspberry Pi 2 (ARMv7) ?

So far I know, is the Pi2 a 32bit system.

best regards 
Georg

**Comment by mkassner** [comment]
For Linux pyuvc will run 32/64bit.

Its just the windows build instructions/support that are 64bit only.

Best,

M

**Comment by sidsapien** [comment]
Hey guys, any update on this issue? I am facing exactly the same issue in Windows. Thanks.

**Comment by fingolfin500** [comment]
I cloned the repo, and checked out an older revision of pupil as suggested on another thread (v0.6.18). This runs from source OK for me on Windows. I had to use 32 bit version of GLFW as well. This revision doesn't use pyuvc from what I understand.

**Comment by imaleev7** [comment]
Hi, has that issue been resolved?  Has anybody actually ever run pupil-labs SW package on Windows?

**Comment by cpicanco** [comment]
It will run, but not with complete functionality. Please, take a look at the wiki for details.

**Comment by imaleev7** [comment]
Hm.. sorry, could you provide a bit more details than just a vague reference to wiki??

specifically - looking through https://github.com/pupil-labs/pyuvc - l;ooks like it only supports Linux and "Darwin" [OS?]   When I try to run python setup.py install, I get "not implemented" error

- when I build package from git clone https://github.com/pupil-labs/libuvc, I'm able to [after modifying CMakeLists.txt to reflect library locations]. However, I still get that error about lack of uvc module. How to I fix that??   

Thanks for all the help.

**Comment by mkassner** [comment]
The issue is iso transfers on windows. Without those implemented in `libusb` `libuvc` will not work on windows.

**Comment by imaleev7** [comment]
Ok, thanks..  so is there a way to run pupil-labs SW on Windows?  Find an older version?  Why not use standard Windows USB / camera driver?

**Comment by imaleev7** [comment]
..for now I gave up on trying to run current source code on Windows. I found version 0.6, and used that for my evaluation. One thing that I immediately noticed is the extreme sensitivity to glints. I realize it's a different topic, and perhaps this message should be moved to a different thread. However - would it be possible to clarify if pupil-labs is designed to work with glints [reflections], and also with remote eye-tracking camera, where FOV may include objects beyond single eye? Thank you.
